plugins/pulp_rpm/plugins/importers/yum/parse/rpm.py

```diff
@@ -143,3 +145,45 @@ def package_signature(headers):
         return None
     # gpg program uses the last 8 characters of the fingerprint
     return signature.split()[-1][-8:]
+
+
+def signature_enabled(config):
+    """
+    Check if the signature policy is enabled.
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :return: true if enabled, false otherwise
+    :rtype: boolean
+    """
+
+    allow_unsigned = config.get("allow_unsigned", True)
+    allow_keys = config.get("allow_keys", set())
+    if not allow_unsigned or allow_keys:
+        return True
+    return False
+
+
+def verify_signature(unit, config):
+    """
+    Verify package signature.
+
+    :param unit: model instance of the package
+    :type  unit: pulp_rpm.plugins.db.models.RPM/DRPM/SRPM
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :raise: PulpCodedException if the signature check did not pass
+    """
+
+    signature = unit.signature
+    allow_unsigned = config.get("allow_unsigned", True)
+    allow_keys = config.get("allow_keys", set())
+    if not allow_unsigned and not signature:
+        raise PulpCodedException(error_code=error_codes.RPM1010, package=unit.filename)
+    if allow_keys:
+        if signature and signature not in allow_keys:
```

Comment:
I find this surprising. Is the ``signature`` variable holding the identifier for a key that was used to create a signature? Or is it the value of an actual signature?

If it's the former, I can understand how this works, but I think the field should be renamed to something like ``signing_key``.